IdentityCore/src/cache/serializers/MSIDKeyedArchiverSerializer.m

```diff
@@ -59,13 +59,21 @@ - (MSIDLegacyTokenCacheItem *)deserialize:(NSData *)data className:(Class)classN
     {
         return nil;
     }
-    
-    NSKeyedUnarchiver *unarchiver = [[NSKeyedUnarchiver alloc] initForReadingWithData:data];
-    // Maintain backward compatibility with ADAL.
-    [unarchiver setClass:MSIDUserInformation.class forClassName:@"ADUserInformation"];
-    [unarchiver setClass:className forClassName:@"ADTokenCacheStoreItem"];
-    MSIDLegacyTokenCacheItem *token = [unarchiver decodeObjectOfClass:className forKey:NSKeyedArchiveRootObjectKey];
-    [unarchiver finishDecoding];
+    MSIDLegacyTokenCacheItem *token = nil;
+    @try
+    {
+        NSKeyedUnarchiver *unarchiver = [[NSKeyedUnarchiver alloc] initForReadingWithData:data];
```

Comment:
initForReadingWithData seems to be deprecated : https://developer.apple.com/documentation/foundation/nskeyedunarchiver/1410862-initforreadingwithdata . Can we update code to use https://developer.apple.com/documentation/foundation/nskeyedunarchiver/2962883-initforreadingfromdata? Maybe add a TODO or try using initForReadingFromData in catch block to see if that worked

Reply:
I can't repro the crash locally, so it'd be costly to build a custom version of Outlook and ask the customer to test it. I think the common core in MSAL already switched to the new method, but here it's for ADAL and we are supposed to deprecate it, so I will leave it as is for now. 
The new catch block would be able to prevent the reported crash.